source/lib.ts

```diff
@@ -318,47 +335,29 @@ const rateLimit = (
 			const maxHits = await retrieveQuota
 			config.validations.max(maxHits)
 
-			// Set the rate limit information on the augmented request object
-			augmentedRequest[config.requestPropertyName] = {
+			const info: RateLimitInfo = {
 				limit: maxHits,
 				current: totalHits,
 				remaining: Math.max(maxHits - totalHits, 0),
 				resetTime,
 			}
 
+			// Set the rate limit information on the augmented request object
+			augmentedRequest[config.requestPropertyName] = info
+
 			// Set the X-RateLimit headers on the response object if enabled
 			if (config.legacyHeaders && !response.headersSent) {
-				response.setHeader('X-RateLimit-Limit', maxHits)
-				response.setHeader(
-					'X-RateLimit-Remaining',
-					augmentedRequest[config.requestPropertyName].remaining,
-				)
-
-				// If we have a resetTime, also provide the current date to help avoid
-				// issues with incorrect clocks.
-				if (resetTime instanceof Date) {
-					response.setHeader('Date', new Date().toUTCString())
-					response.setHeader(
-						'X-RateLimit-Reset',
-						Math.ceil(resetTime.getTime() / 1000),
-					)
-				}
+				setLegacyHeaders(response, info)
 			}
 
 			// Set the standardized RateLimit headers on the response object
 			// if enabled.
 			if (config.standardHeaders && !response.headersSent) {
-				response.setHeader('RateLimit-Limit', maxHits)
-				response.setHeader(
-					'RateLimit-Remaining',
-					augmentedRequest[config.requestPropertyName].remaining,
-				)
-
-				if (resetTime) {
-					const deltaSeconds = Math.ceil(
-						(resetTime.getTime() - Date.now()) / 1000,
-					)
-					response.setHeader('RateLimit-Reset', Math.max(0, deltaSeconds))
+				if (config.standardHeaders === 'draft-6') {
+					setStandardHeadersDraft6(response, info)
+				} else if (config.standardHeaders === 'draft-7') {
+					// Todo: validation check to ensure a reset time is provided by the store
```

Comment:
The spec now requires a reset time to be provided, which we can't do with some stores. I want to add a validation warning for that situation here, to at least make folks aware of it.